testing/e2e/e2e_api_test.go

```diff
@@ -732,3 +721,149 @@ func (s *BeaconKitE2ESuite) TestGetValidatorBalancesForGenesis() {
 		s.Require().True(balance.Balance <= 32e9, "Validator balance should not exceed 32 BERA")
 	}
 }
+
+// getValidator gets the validator with the given stateID and optional options.
+func (s *BeaconKitE2ESuite) getValidator(stateID string, options ...map[string]string) (*http.Response, error) {
+	client := s.initHTTPBeaconTest()
+
+	url := fmt.Sprintf("/eth/v1/beacon/states/%s/validators", stateID)
+
+	// Process options if provided
+	if len(options) > 0 {
+		queryParams := []string{}
+
+		// Extract options from the map
+		for _, opts := range options {
+			for key, value := range opts {
+				queryParams = append(queryParams, fmt.Sprintf("%s=%s", key, value))
+			}
+		}
+
+		// Add query parameters to URL if any exist
+		if len(queryParams) > 0 {
+			url = url + "?" + strings.Join(queryParams, "&")
+		}
+	}
+
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get validator: %w", err)
+	}
+	if resp == nil {
+		return nil, errors.New("received nil response")
+	}
+	return resp, nil
+}
+
+// decodeValidatorsResponse decodes a response containing multiple validators.
+func (s *BeaconKitE2ESuite) decodeValidatorsResponse(resp *http.Response) (*[]beacontypes.ValidatorData, error) {
+	validators, err := decodeResponse[[]beacontypes.ValidatorData](resp)
+	if err != nil {
+		return nil, err
+	}
+	return &validators, nil
+}
+
+// TestGetValidatorsWithStateHead tests querying validators with state head.
+func (s *BeaconKitE2ESuite) TestGetValidatorsWithStateHead() {
+	resp, err := s.getValidator(utils.StateIDHead)
+	s.Require().NoError(err)
+	s.Require().Equal(http.StatusOK, resp.StatusCode)
+	defer resp.Body.Close()
+
+	validators, err := s.decodeValidatorsResponse(resp)
+	s.Require().NoError(err)
+	s.Require().NotNil(validators)
+
+	for _, validator := range *validators {
+		s.Require().True(validator.Status == "active_ongoing")
+		s.Require().True(validator.Balance > 0, "Validator balance should be positive")
+		// 4e12 Gwei = 4 * 10^12 Gwei = 4,000,000,000,000 Gwei = 4000 BERA
+		s.Require().True(validator.Balance <= 4e12, "Validator balance should not exceed 4000 BERA")
+	}
+}
+
+// TestGetValidatorsWithID tests querying validators with ID parameter.
+func (s *BeaconKitE2ESuite) TestGetValidatorsWithID() {
+	resp, err := s.getValidator(utils.StateIDHead, map[string]string{
+		"id": "0",
+	})
```

Comment:
confused since couldn't this just be replaced by calling `getStateValidator("head", "0")` ? Or are you specifically testing using the HTTP query params here?